src/gtest/test_deprecation.cpp

```diff
@@ -102,4 +102,25 @@ TEST_F(DeprecationTest, DeprecatedNodeKeepsRunningIfCurrentVersionDisabled) {
     EXPECT_CALL(mock_, ThreadSafeMessageBox(::testing::_, "", CClientUIInterface::MSG_ERROR));
     EnforceNodeDeprecation(DEPRECATION_HEIGHT);
     EXPECT_FALSE(ShutdownRequested());
+}
+
+TEST_F(DeprecationTest, DeprecatedNodeIgnoredOnTestnet) {
+    EXPECT_FALSE(ShutdownRequested());
+    mapArgs["-testnet"] = "1";
+    EnforceNodeDeprecation(DEPRECATION_HEIGHT+1);
+    EXPECT_FALSE(ShutdownRequested());
+}
+
+TEST_F(DeprecationTest, DeprecatedNodeIgnoredOnRegtest) {
+    EXPECT_FALSE(ShutdownRequested());
+    mapArgs["-regtest"] = "1";
+    EnforceNodeDeprecation(DEPRECATION_HEIGHT+1);
+    EXPECT_FALSE(ShutdownRequested());
+}
+
+TEST_F(DeprecationTest, DeprecatedNodeHonoredOnMainnet) {
+    EXPECT_FALSE(ShutdownRequested());
+    mapArgs["-mainnet"] = "1";
```

Comment:
There is no `-mainnet` config flag. Remove this test.

Reply:
Ah, my mistake. I see now that any jibberish flag can be passed and is simply ignored, will remove.